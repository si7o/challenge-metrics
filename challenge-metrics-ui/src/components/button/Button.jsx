import React from 'react'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import './styles.scss'

const Button = (props) => {
  const { title, className, onClick, children, primary, disabled, type } = props

  const handleClick = (e) => {
    onClick(e)
  }

  const classname = clsx('button', className, { primary })

  return (
    <button type={type} title={title} onClick={handleClick} className={classname} disabled={disabled}>
      {children}
    </button>
  )
}

Button.propTypes = {
  title: PropTypes.string,
  className: PropTypes.any,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['button', 'submit', 'reset'])
}

Button.defaultProps = {
  onClick: () => { },
  type: 'button'
}

export default Button
