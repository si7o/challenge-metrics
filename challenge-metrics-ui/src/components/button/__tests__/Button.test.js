import React from 'react'
import { render, screen } from '@testing-library/react'

import Button from '../Button'
import userEvent from '@testing-library/user-event'

describe('<Button>', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  test('renders default', () => {
    const { container } = render(<Button>test button</Button>)

    expect(container).toMatchSnapshot()
  })

  test('when clicks, triggers onClick', () => {
    const mockedOnClick = jest.fn()

    render(
      <Button onClick={mockedOnClick}>test button</Button>
    )

    userEvent.click(screen.getByRole('button'))

    expect(mockedOnClick).toHaveBeenCalled()
  })

  test('renders disabled', () => {
    render(<Button disabled>test button</Button>)

    expect(screen.getByRole('button')).toHaveAttribute('disabled')
  })
})
