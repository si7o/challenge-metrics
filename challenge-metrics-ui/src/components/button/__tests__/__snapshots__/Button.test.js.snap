// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders default 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    test button
  </button>
</div>
`;
